libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Recognizers/CrossTrainedRecognizerSet.cs

```diff
@@ -0,0 +1,221 @@
+            }
+
+            if (text == null)
+            {
+                throw new ArgumentNullException(nameof(text));
+            }
+
+            EnsureRecognizerIds();
+
+            // run all of the recognizers in parallel
+            var results = await Task.WhenAll(Recognizers.Select(r => r.RecognizeAsync(dialogContext, text, locale, cancellationToken)));
+
+            return ProcessResults(results);
+        }
+
+        private RecognizerResult ProcessResults(RecognizerResult[] results)
+        {
+            // put results into a dictionary for easier lookup while processing.
+            Dictionary<string, RecognizerResult> recognizerResults = new Dictionary<string, RecognizerResult>(System.StringComparer.OrdinalIgnoreCase);
+            Dictionary<string, string> intents = new Dictionary<string, string>(System.StringComparer.OrdinalIgnoreCase);
+            
+            string text = string.Empty;
+            for (int iRecognizer = 0; iRecognizer < Recognizers.Count; iRecognizer++)
+            {
+                var recognizer = Recognizers[iRecognizer];
+                var result = results[iRecognizer];
+                recognizerResults[recognizer.Id] = result;
+                var (topIntent, score) = result.GetTopScoringIntent();
+                intents[recognizer.Id] = topIntent;
+                text = result.Text ?? string.Empty;
+            }
+
+            // this is the consensusRecognizer to use
+            string consensusRecognizerId = null;
+            foreach (var recognizer in Recognizers)
+            {
+                var intent = intents[recognizer.Id];
+
+                if (!IsRedirect(intent))
+                {
+                    // we have a real intent and it's the first one we found.
+                    if (consensusRecognizerId == null)
+                    {
+                        consensusRecognizerId = recognizer.Id;
+                    }
+                    else
+                    {
+                        // we have a second recognizer with an intent
+
+                        // if one of them is None intent, then go with the other one.
+                        if (intent == NoneIntent)
+                        {
+                            // then we are fine with the one we have, just ignore this one
+                            continue;
+                        }
+                        else if (intents[consensusRecognizerId] == "None")
+                        {
+                            // then we can drop the old one and go with the new one instead
+                            consensusRecognizerId = recognizer.Id;
+                        }
+                        else
+                        {
+                            // ambigious because of 2 real intents, and neither are None so return AmbigiousIntent
+                            return CreateAmbigiousIntentResult(text, recognizerResults);
```

Comment:
>CreateAmbigiousIntentResult [](start = 35, length = 27)

Shouldn't you support more than 2?  What happens if you have three recognizers and they all want to handle?